index.js

```diff
@@ -0,0 +1,122 @@
+            })).sort((a, b) => a.sort - b.sort).map(a => a.value)
+            let shuffledArr = shuffle(squares)
+            //adding each array element to html of container
+            shuffledArr.forEach((square) => {
+                document.getElementById('gridContainer').innerHTML += square
+            })
+        }
+
+    //validates form input
+    if(!validateInput(gridSize, numLynx)){
+        document.getElementById('errorMessage').textContent = errorMessage
+    }else {
+        //execute function
+        generateSquares()
+        turnCard()
+        //sets the grid container display to default
+        document.getElementById('gameTitle').style.display = ""
+        document.getElementById('gridContainer').style.display = ""
+        //scroll to top of game play area
+        document.querySelector('#gameTitle').scrollIntoView({
+            behavior: 'smooth'
+        })
+    }
+})
+
+
+// DOCBLOCK READ PLS
+// Function takes 2 integer inputs from form.
+//num1 is grid size.
+//num2 is the number of lynx.
+//Function checks if number is integer,
+// if positive
+// and if hitboxes (num2) does not exceed half of the grid size (num1).
+// global error message is populated upon error, dictated by the conditional, ready to be displayed
+//return true if no errors
+
+let errorMessage = ""
+
+function validateInput(num1, num2) {
+    if (!Number.isInteger(num1) || !Number.isInteger(num2)){
+        errorMessage = 'Please only input whole numbers'
+        return false
+    }
+    if (num1 > 100 || num2 > 50) {
+        errorMessage = 'Grid size must not exceed 100 and number of lynx must not exceed 50'
+        return false
+    }
+    if (Math.sign(num1) === 0 || Math.sign(num2) === 0) {
+        errorMessage = 'Can not input 0'
+        return false
+    }
+    if (Math.sign(num1) === -1 || Math.sign(num2) === -1) {
+        errorMessage = 'You must have a positive grid size and number of lynx to find'
+        return false
+    }
+    if ((num1 / 2) < num2) {
+        errorMessage = 'Number of lynx can not exceed half of the number you picked for the Forest size'
+        return false
+    }
+
+    return true
+}
+
+function turnCard() {
+// Grabs all grid-item divs as gridItems to be used in forEach
+let gridItems = document.querySelectorAll('.grid-item')
+// DOCBLOCK BELOW
+// Does a forEach over gridItems and checks the data-hit value for each of them
+// When clicked, if data-hit is 1: Applies class='hit' and changes textContent to 'HIT'
+// When clicked, if data-hit is 0: Applies class='miss' and changes textContent to 'MISS'
+// Both .hit and .miss are styled in CSS, ready for application.
+// DOCBLOCK ABOVE
+
+gridItems.forEach( (item) => {
+    item.addEventListener('click', () => {
+        if(item.dataset.hit === '1'){
+            item.style.backgroundImage = "url('project-assets/babylynx2.jpg')"
+            item.textContent = 'Hooray, you\'ve found a lynx!'
+        }
+        if(item.dataset.hit === '0'){
```

Comment:
why not use an else here? there isn't any other value that hit could be so no need to check it again